### PR TITLE
dummyfs: Remove NULL check from read/write

### PR DESCRIPTION
**Pull request by agkaminski**
There is zero reason for us to check for NULL buffer - as it is not possible to receive buff == NULL and len != 0 from msg. Regarding use outside of standard server - there is no reason to pass same parameter combo, so there's no need to check for it. There many other invalid pointers apart from NULL, we can't check for them all anyway (other than try and crash).

DONE: RTOS-353

## Motivation and Context

Fixes https://github.com/phoenix-rtos/phoenix-rtos-project/issues/598

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: dummyfs/dummyfs.c [hunk 1]
```diff
@@ -681,29 +681,18 @@ int dummyfs_read(void *ctx, oid_t *oid, offs_t offs, char *buff, size_t len)
...
-		ret = -EINVAL;
-
-	if (buff == NULL)
-		ret = -EINVAL;
-
-	if (o->size <= offs) {
-		object_put(fs, o);
-		return 0;
+	if (o == NULL) {
+		return -ENOENT;
```
Comment (Maxez) [inline dummyfs/dummyfs.c:685]: I'd leave `EINVAL` here. `read()` operation doesn't define `ENOENT` as a valid errno.

File: dummyfs/dummyfs.c [hunk 2]
```diff
@@ -745,19 +736,13 @@ int dummyfs_write(void *ctx, oid_t *oid, offs_t offs, const char *buff, size_t l
 	int ret = EOK;
 
 	o = object_get(fs, oid->id);
+	if (o == NULL) {
+		return -ENOENT;
```
Comment (Maxez) [inline dummyfs/dummyfs.c:740]: Same as above. Return `-EINVAL` instead of `-ENOENT`.

**Review by Maxez — CHANGES_REQUESTED** [review]

File: dummyfs/dummyfs.c (hunk 1 above)
Comment (agkaminski) [reply dummyfs/dummyfs.c:685]: On the other hand such situation within read()/write() regime shouldn't be even possible - to access non-existing oid via existing fd ;) Anyway, I can change it back

Comment (Maxez) [reply dummyfs/dummyfs.c]: I agree, `o == NULL` shouldn't ever happen for a valid fd. Nonetheless I'd stick to `-EINVAL`.

**Review by Maxez — APPROVED** [review]
LGTM.

**Pull request merged**